backend/src/packages/chaiNNer_onnx/onnx/processing/upscale_image.py

```diff
@@ -96,9 +111,12 @@ def upscale_image_node(
     h, w, c = get_h_w_c(img)
     logger.debug(f"Image is {h}x{w}x{c}")
 
+    should_separate_alpha = separate_alpha and c == 4
```

Comment:
Why make this variable? `convenient_upscale` will ignore the parameter unless it's applicable anyway, so you don't need to make any checks yourself.